core/src/Kokkos_Macros.hpp

```diff
@@ -543,9 +572,13 @@
 // If compiling with CUDA, we must use relocateable device code
 // to enable the task policy.
 
-#if !defined(KOKKOS_ENABLE_CUDA) || \
+#if defined(KOKKOS_ENABLE_CUDA) && \
     defined(KOKKOS_ENABLE_CUDA_RELOCATABLE_DEVICE_CODE)
 #define KOKKOS_ENABLE_TASKDAG
+#else
```

Comment:
This logic is wrong because it turns Tasking for CUDA now on if `KOKKOS_ENABLE_CUDA_RELOCATABLE_DEVICE_CODE` is off.

Reply:
```
#if defined(KOKKOS_ENABLE_CUDA)
#if defined(KOKKOS_ENABLE_CUDA_RELOCATABLE_DEVICE_CODE)
#define KOKKOS_ENABLE_TASKDAG
#endif
#else
#ifndef KOKKOS_ENABLE_HIP
#define KOKKOS_ENABLE_TASKDAG
#endif
#endif
```

Reply:
gonna push this in a sec